src/main/mondrian/rolap/RolapMemberBase.java

```diff
@@ -155,6 +155,18 @@ public RolapMember getParentMember() {
         return (RolapMember) super.getParentMember();
     }
 
+    @Override
+    public Object getCaptionValue() {
+        if (captionValue != null) {
+            return captionValue;
+        }
+        return getKey();
```

Comment:
I think the current behavior is that if captionColumn is not defined, we fall back first to name, then to finally to key.